### DISCUSSION TITLE
【keyball44】Pro Microが認識されない

### DISCUSSION THREAD

**Discussion by nonagase** [Yowkees/keyball#682] [Q&A] [answered]
自作キーボードが初めての為、基礎的なことを聞いていればすみません。
keyball44組立しており「ProMicroの書き込みと動作確認」を実施していますが、片方のPro Microがファームウェアの書き込みで「互換性のあるデバイスは見つかりません」となり、書き込みができません。

基盤と接続せずに、Pro Microのみでusbで繋いでも「互換性のあるデバイスは見つかりません」となります。
ショートさせてリセットしても結果は変わらずとなります。

外因の切り分けや、対応方法について何かわかることがありましたらご教示お願いします。
コンスルーの向きを揃えずにハンダ付けしてしまったため、片方のコンスルーのみ一度ハンダを取り除いて新しいコンスルーを付け直しております。
![IMG_7387](https://github.com/user-attachments/assets/08642d56-71d9-406d-a7c1-5ad8932f5683)

![IMG_7388](https://github.com/user-attachments/assets/8645dabc-87af-48b9-b58a-319a47a88c54)

![IMG_7389](https://github.com/user-attachments/assets/1b0a6232-751c-4691-b301-c168d71c2647)

![IMG_7390](https://github.com/user-attachments/assets/93058ed1-3ad8-404c-99c5-9dbc1881f8b9)

**Comment by Yowkees** [comment, marked as answer]
ご連絡ありがとうございます。

半田付けについては隣とショートしているものは見当たらないので大丈夫だと思います。
（ただコンスルーで隠れた部分が半田屑でショートしているようですと、交換した方が良いかもしれません）

ProMicroのリセットですがコツがありまして、リセットを素早く２回押す必要があります。USB接続した状態で、うまく２回RSTとCNGをショートさせると「ポロロン」とデバイス認識音がすると思います。

お手数ですがご確認お願いいたします。

**Comment by nonagase** [comment]
ご回答ありがとうございます。
回答の通り2回連続でショートさせることで、解決いたしました。
キーも全て反応する事が確認出来ましたので次の工程に進めます。

**Comment by Yowkees** [comment]
無事に動くようになったようで良かったです！
今後ともどうぞよろしくお願いいたします。
